hadoop-common-project/hadoop-common/src/main/resources/core-default.xml

```diff
@@ -1344,34 +1338,34 @@
   </description>
 </property>
 
+
 <property>
-  <name>fs.s3a.retry.limit</name>
-  <value>4</value>
-  <description>
-    Number of times to retry any repeatable S3 client request on failure,
-    excluding throttling requests.
-  </description>
+  <name>fs.s3a.attempts.maximum</name>
+  <value>20</value>
+  <description>How many times we should retry commands on transient errors,
+  excluding throttling errors.</description>
```

Comment:
Interesting.  One of my concerns about all the retry logic being added here is that it is an invasive change and I'm feeling like there might be unintended consequences somewhere.  I've been thinking that making it more configurable would mitigate the risk.. I'd lean towards making more types/classes of retry configuration instead of fewer.   For example here, I'd like to have SDK retries configured separately.  I mentioned before also the idea of having another retry policy for riskier parts (e.g. delete).  Thoughts?

Reply:
I worry about overconfig of things which aren't easy to test. At the same time, the bulk/recursive ops are special cases because we need to think "how best to retry?". Is it per-op, or would you want to step back and start from scratch. With the fact that delete retries aren't isolated from other ops in the bucket, that's one where we could make a case for saying "no, no retries except in higher level algorithms (like the commit one)"